src/Compilers/CSharp/Test/Emit/PDB/CSharpDeterministicBuildCompilationTests.cs

```diff
@@ -74,11 +82,15 @@ private static void TestDeterministicCompilationCSharp(string langVersion, Synta
                 using (var embeddedPdb = peReader.ReadEmbeddedPortablePdbDebugDirectoryData(embedded))
                 {
                     var pdbReader = embeddedPdb.GetMetadataReader();
-
                     var metadataReferenceReader = DeterministicBuildCompilationTestHelpers.GetSingleBlob(PortableCustomDebugInfoKinds.CompilationMetadataReferences, pdbReader);
                     var compilationOptionsReader = DeterministicBuildCompilationTestHelpers.GetSingleBlob(PortableCustomDebugInfoKinds.CompilationOptions, pdbReader);
 
-                    VerifyCompilationOptions(compilationOptions, originalCompilation, emitOptions, compilationOptionsReader, langVersion);
+                    if (debugDocumentsCount is not null)
+                    {
+                        Assert.Equal(debugDocumentsCount, pdbReader.Documents.Count);
```

Comment:
>Assert.Equal(debugDocumentsCount, pdbReader.Documents.Count); [](start = 24, length = 61)

Should we use check this assert unconditionally? `Assert.Equal(debugDocumentsCount ?? syntaxTrees.Length, pdbReader.Documents.Count);`

Reply:
We could if we ensure every test that doesn't pass 'debugDocumentsCount' passes a unique path of each of its documents, which isn't hard to do.